events/views.py

```diff
@@ -4,13 +4,46 @@
 from django.urls import reverse
 from .models import Event, Location
 from django.contrib.auth.decorators import login_required
+from django.utils import timezone
+from .forms import EventFilterForm
 
 # Create your views here.
 
 
 def index(request):
-    events = Event.objects.filter(is_active=True)
-    return render(request, "events/events.html", {"events": events})
+    events = Event.objects.filter(end_time__gt=timezone.now(), is_active=True).order_by(
+        "-start_time"
+    )
+
+    # Process the filter form
+    if request.GET:
+        form = EventFilterForm(request.GET)
+        if form.is_valid():
+            if form.cleaned_data["start_time"]:
+                events = events.filter(start_time__gte=form.cleaned_data["start_time"])
+            if form.cleaned_data["end_time"]:
+                if form.cleaned_data["end_time"] <= timezone.now():
```

Comment:
We should add these checks on frontend as well?

Reply:
added the checks on frontend.